### PR TITLE
GetPendingModeratorRequestsPaginatedDone

### PR DESCRIPTION
**Pull request by aabdoo23**
#### PR Classification
Code refactor and feature enhancement.
Too many changes because it is based on the previous branch not origin
#### PR Summary
Refactored repository classes and added new methods, updated service configurations, and added migrations.
- `BaseUserRepository`: Constructor now throws `ArgumentNullException` if `SchedsForumsDbContext` is null.
- `BaseRepository`: Refactored to use parameterized constructor; added `GetTotalCount` and `GetFromTo` methods.
- `ServiceCollectionExtensions`: Updated repository registrations and refactored JWT configuration methods.
- `SchedsForums.Infrastructure.csproj`: Added new `Migrations` folder.

### CODE REVIEW

File: SchedsForums.Api/Controllers/AdminController.cs [hunk 1]
```diff
@@ -19,5 +20,12 @@ public async Task<IActionResult> CreateAdminAsync([FromBody] CreateAdminCommand
             var result = await _mediator.Send(command);
             return Ok(result);
         }
+
+        [HttpGet("PendingModerators")]
+        public async Task<IActionResult> GetModeratorSignUpRequestsAsync([FromBody] GetPendingModeratorsQuery query)
```
Comment (MostafaHgz) [inline SchedsForums.Api/Controllers/AdminController.cs]: Thishould be in a new controller

File: SchedsForums.Api/appsettings.Development.json [hunk 2]
```diff
@@ -5,8 +5,8 @@
       "Microsoft.AspNetCore": "Warning"
     }
   },
-  "JWT": {
```
Comment (MostafaHgz) [inline SchedsForums.Api/appsettings.Development.json]: Pull from master to update this branch

File: SchedsForums.Application/Interfaces/Common/IBaseRepository.cs [hunk 3]
```diff
@@ -5,9 +5,12 @@ namespace SchedsForums.Application.Interfaces.Common
     public interface IBaseRepository<T> where T : BaseEntity
     {
         Task<T?> GetByIdAsync(string id);
+        Task<T?> GetByIdAsync(Guid id);
         Task<IEnumerable<T>> GetAllAsync();
         Task<T> InsertAsync(T entity);
         Task<T> UpdateAsync(T entity);
-        Task DeleteAsync(string id);
+        Task DeleteAsync(Guid id);
+        Task<IEnumerable<T>> GetFromTo(int pageNumber, int pageSize);
```
Comment (MostafaHgz) [inline SchedsForums.Application/Interfaces/Common/IBaseRepository.cs]: Renamed to get paginated

File: SchedsForums.Application/Queries/Admins/GetPendingModerators/GetPendingModeratorsQueryHandler.cs [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            ?? throw new ArgumentNullException(nameof(repository));
+
+        public async Task<GetPendingModeratorsQueryResponseDTO> Handle(
+            GetPendingModeratorsQuery request,
+            CancellationToken cancellationToken)
+        {
+            var pendingModeratorsQueryResult = await _pendingModeratorsRepository.GetFromTo(request.PageNumber, request.PageSize);
+            var totalCount = await _pendingModeratorsRepository.GetTotalCount();
+
+            return new GetPendingModeratorsQueryResponseDTO
```
Comment (MostafaHgz) [inline SchedsForums.Application/Queries/Admins/GetPendingModerators/GetPendingModeratorsQueryHandler.cs]: Create a single paginted bas response

Comment (MostafaHgz) [reply SchedsForums.Application/Queries/Admins/GetPendingModerators/GetPendingModeratorsQueryHandler.cs]: Paginated response should return from the query not here as it's common in all queries

**Review by MostafaHgz — CHANGES_REQUESTED** [review]

File: SchedsForums.Api/appsettings.Development.json (hunk 2 above)
Comment (aabdoo23) [reply SchedsForums.Api/appsettings.Development.json]: updated

File: SchedsForums.Application/Interfaces/Common/DTOs/BaseResponseDTO.cs [hunk 5]
```diff
@@ -0,0 +1,7 @@
+﻿namespace SchedsForums.Application.Interfaces.Common.DTOs
+{
+    public class BaseResponseDTO
```
Comment (MostafaHgz) [inline SchedsForums.Application/Interfaces/Common/DTOs/BaseResponseDTO.cs]: rename to BaseIdResponseDTO

File: SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedDTO.cs [hunk 6]
```diff
@@ -0,0 +1,11 @@
+﻿namespace SchedsForums.Application.Queries.Common.DTOs
+{
+    public abstract class BaseGetPaginatedDTO<T>
```
Comment (MostafaHgz) [inline SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedDTO.cs]: rename to PaginatedResponseDTO

File: SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedDTO.cs [hunk 7]
```diff
@@ -0,0 +1,11 @@
...
+    {
+        public int PageNumber { get; set; }
+        public int PageSize { get; set; }
+        public int TotalCount { get; set; }
+        public int ReturnedCount { get; set; }
```
Comment (MostafaHgz) [inline SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedDTO.cs]: no need for this

File: SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedDTO.cs [hunk 8]
```diff
@@ -0,0 +1,11 @@
...
+        public IEnumerable<T> Data { get; set; }
```
Comment (MostafaHgz) [inline SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedDTO.cs]: this should be IList<>

File: SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedEntityDTO.cs [hunk 9]
```diff
@@ -0,0 +1,8 @@
+﻿using SchedsForums.Domain.Entities.Common;
+
+namespace SchedsForums.Application.Queries.Common.DTOs
```
Comment (MostafaHgz) [inline SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedEntityDTO.cs]: what does this do

Comment (MostafaHgz) [reply SchedsForums.Application/Queries/Common/DTOs/BaseGetPaginatedEntityDTO.cs]: this should be in the repository

File: SchedsForums.Infrastructure/Repositories/PendingModeratorRepository.cs [hunk 10]
```diff
@@ -0,0 +1,38 @@
...
+            {
+                Data = entityResponse.Data.Select(pm => new BasePendingModeratorResponseDTO(pm)),
+                ReturnedCount = entityResponse.ReturnedCount,
+                TotalCount = entityResponse.TotalCount,
+                PageNumber = entityResponse.PageNumber,
+                PageSize = entityResponse.PageSize
+            };
+        }
+
+        public override async Task<IEnumerable<PendingModerator>> GetAllAsync()
```
Comment (MostafaHgz) [inline SchedsForums.Infrastructure/Repositories/PendingModeratorRepository.cs]: remove this

**Review by MostafaHgz — CHANGES_REQUESTED** [review]

File: SchedsForums.Application/Queries/Common/DTOs/PaginatedEntityDTO.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+﻿using SchedsForums.Domain.Entities.Common;
+
+namespace SchedsForums.Application.Queries.Common.DTOs
```
Comment (MostafaHgz) [inline SchedsForums.Application/Queries/Common/DTOs/PaginatedEntityDTO.cs:3]: I believe we agreed this should be moved to the infrastructure

File: SchedsForums.Infrastructure/Repositories/Common/BaseRepository.cs [hunk 12]
```diff
@@ -36,11 +33,36 @@ public async Task<IEnumerable<T>> GetAllAsync()
...
+        public virtual async Task<PaginatedEntityDTO<T>> GetPaginatedContentAsync(IQueryable<T> queryable, int pageNumber, int pageSize)
+        {
+            var totalCount = await queryable.CountAsync();
+
+            var data = await queryable
+                .OrderBy(e => e.CreatedAt)
+                .Skip((pageNumber - 1) * pageSize)
+                .Take(pageSize)
+                .ToListAsync();
+            var returnedCount = data.Count;
```
Comment (MostafaHgz) [inline SchedsForums.Infrastructure/Repositories/Common/BaseRepository.cs]: this is not used anymore

**Review by MostafaHgz — CHANGES_REQUESTED** [review]

**Review by MostafaHgz — APPROVED** [review]

**Pull request merged**